client/lib/mention.js

```diff
@@ -0,0 +1,71 @@
+var hueHash = require('./hue-hash')
+
+/**
+ * Determines if name contains the characters in the partial name, in order.
+ */
+module.exports.containsSubseq = function(name, part) {
+  var offset = 0
+  var remain = part
+  var nexdex
+  while (remain !== "") {
+    nexdex = name.indexOf(remain.substr(0, 1), offset)
+    if (nexdex < 0) {
+      return false
+    }
+    offset = nexdex + 1
+    remain = remain.substr(1)
```

Comment:
A bit of an over-optimization: it'd probably be simpler to step an index var forward instead of creating a new string with every `substr` call.